tests/test_dataset.py

```diff
@@ -1201,6 +1201,34 @@ def test_sub_dataset(ds1, ds2):
     assert np.all(ds3[1].to_numpy() == 1.8)
 
 
+def test_multiply_dataset(ds1, ds2):
+    dsa = mikeio.Dataset({"Foo": mikeio.DataArray([1, 2, 3])})
+    dsb = mikeio.Dataset({"Foo": mikeio.DataArray([4, 5, 6])})
+    dsr = dsa * dsb
+    assert np.all(dsr[0].to_numpy() == np.array([4, 10, 18]))
+
+
+def test_multiply_datasets_must_match():
+    dsa = mikeio.Dataset({"Foo": mikeio.DataArray([1, 2, 3])})
+    dsb = mikeio.Dataset({"Bar": mikeio.DataArray([4, 5, 6])})
+    with pytest.raises(ValueError):
```

Comment:
Why do the name/item need to match on multiplication/division?

Reply:
Let's try with minimal requirements. If two datasets have the same number of items, and the shapes of the data is identical, we can multiply them.

Reply:
Additional thought: could this feature also be used for scaling several items in a file with the same spatially/temporally varying field? 

a) Should we allow that shapes conform but items in "other" is either the same as "this" or 1 (meaning same scaling applied to all variables).

b) Even further should we allow that as long as broadcasting rules allow multiplication then shape of "other" is okay? (so I could e.g. scale a dfsu with a dfs0)

Reply:
If we decide to go down this rabbit hole, I think we should consider restructuring MIKE IO to rely on xarray instead of numpy and to make use of the broadcasting in xarray. https://docs.xarray.dev/en/stable/user-guide/computation.html#broadcasting-by-dimension-name

Reply:
Let's wait until someone has the need